table_engine/src/partition/mod.rs

```diff
@@ -2,26 +2,84 @@
 
 //! Partitioned table supports
 
-mod expression;
 pub mod rule;
 
-/// Partition type of table
+use proto::meta_update as meta_pb;
+
+/// Info for how to partition table
+#[derive(Clone, Debug, PartialEq)]
+pub enum PartitionInfo {
+    Hash(HashPartitionInfo),
+}
+
+impl PartitionInfo {
+    pub fn get_definitions(&self) -> Vec<Definition> {
+        match self {
+            Self::Hash(v) => v.definitions.clone(),
+        }
+    }
+}
+
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub enum PartitionType {
-    None = 0,
-    Hash = 1,
+pub struct Definition {
+    pub name: String,
+    pub origin_name: Option<String>,
 }
 
-/// Size type of partition num
-pub type PartitionNum = u16;
+#[derive(Clone, Debug, PartialEq)]
+pub struct HashPartitionInfo {
+    pub definitions: Vec<Definition>,
+    pub columns: Vec<String>,
```

Comment:
I think store `Expr` is enough.
We can call `accept` to get column name in `Expr`.